v3/mqtt.go

```diff
@@ -63,183 +94,204 @@ type (
 	// another interface (and the struct implement the interface if desired).
 	// But this is sufficient for now.
 	MqttDelegate interface {
```

Comment:
I suggest making MqttDelegate interface into a facade interface of smaller interface.  It will allow extendibility in the future.

Reply:
Pushed last night. A few tweaks required elsewhere, but I like how it turned out. I think it might answer some of @takeshinoda 's questions too.